script/upgrades/cCOP/cCOPChecks.verify.sol

```diff
@@ -0,0 +1,438 @@
+
+import { cCOPChecksBase } from "./cCOPChecks.base.sol";
+import { cCOPConfig, Config } from "./Config.sol";
+
+/**
+ * @title IBrokerWithCasts
+ * @notice Interface for Broker with tuple -> struct casting
+ * @dev This is used to access the internal trading limits
+ * config as a struct as opposed to a tuple.
+ */
+interface IBrokerWithCasts {
+  function tradingLimitsConfig(bytes32 id) external view returns (TradingLimits.Config memory);
+}
+
+contract cCOPChecksVerify is cCOPChecksBase {
+  using TradingLimits for TradingLimits.Config;
+
+  uint256 constant PRE_EXISTING_POOLS = 12;
+
+  ICeloGovernance celoGovernance;
+
+  constructor() public {
+    setUp();
+    celoGovernance = ICeloGovernance(governance);
+  }
+
+  function run() public {
+    cCOPConfig.cCOP memory config = cCOPConfig.get(contracts);
+    console.log("\nStarting cCOP checks:");
+
+    console.log("\n==  Rate feeds ==");
+    console.log("   COPUSD: %s", config.rateFeedConfig.rateFeedID);
+
+    verifyToken(config);
+
+    // TODO: Exchange creation is currently commented out.
+    //       Once we have rate feeds, we can uncomment this line
+    // verifyExchange(config);
+    verifyCircuitBreaker(config);
+  }
+
+  function verifyToken(cCOPConfig.cCOP memory config) internal {
+    console.log("\n== Verifying Token Config Transactions ==");
+    verifyOwner();
+    verifyCCOPStableToken(config);
+    verifycCOPAddedToReserve();
+    verifycCOPAddedToFeeCurrencyWhitelist();
+    verifyConstitution();
+  }
+
+  function verifyOwner() internal view {
+    require(Proxy(cCOP)._getOwner() == governance, "StableTokenCOP Proxy ownership not transferred to governance");
+    console.log("🟢 Contract ownerships transferred to governance");
+  }
+
+  function verifyCCOPStableToken(cCOPConfig.cCOP memory config) internal {
+    StableTokenCOPProxy stableTokenCOPProxy = StableTokenCOPProxy(cCOP);
+    address stableTokenV2 = contracts.deployed("StableTokenV2");
+
+    address cCOPImplementation = stableTokenCOPProxy._getImplementation();
+    if (cCOPImplementation != stableTokenV2) {
+      console.log(
+        "The implementation from StableTokenCOPProxy: %s does not match the deployed StableTokenV2 address: %s.",
+        cCOPImplementation,
+        stableTokenV2
+      );
+      revert("StableTokenCOPProxy does not points to StableTokenV2 deployed implementation. See logs.");
+    }
+    console.log("🟢 StableTokenCOPProxy has the correct implementation address");
+
+    // ----- verify initialization parameters -----
+    IStableTokenV2 cCOPToken = IStableTokenV2(cCOP);
+    IERC20Metadata cCOPTokenMetadata = IERC20Metadata(cCOP);
+
+    assertEq(cCOPTokenMetadata.name(), config.stableTokenConfig.name, "❗️❌ cCOP name not set correctly!");
+    assertEq(cCOPTokenMetadata.symbol(), config.stableTokenConfig.symbol, "❗️❌ cCOP symbol not set correctly!");
+    assertEq(cCOPToken.broker(), broker, "❗️❌ cCOP broker not set correctly!");
+
```

Comment:
This was an outstanding TODO, as I was not certain it was something we needed to do. I've since confirmed we don't need to do it, so I won't add the check but will remove the setting in initialize.